library/src/androidTest/java/com/google/android/material/motion/runtime/SchedulerTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2016 The Material Motion Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.material.motion.runtime;
+
+import android.animation.Animator;
+import android.animation.AnimatorListenerAdapter;
+import android.test.AndroidTestCase;
+import android.view.View;
+import android.widget.TextView;
+
+public class SchedulerTest extends AndroidTestCase {
+
+  private Scheduler scheduler;
+  private TextView textView;
+  private Transaction transaction;
+
+  @Override
+  protected void setUp() throws Exception {
+    super.setUp();
+    scheduler = new Scheduler();
+    textView = new TextView(getContext());
+    transaction = new Transaction();
+  }
+
+  public void testInitialSchedulerState() {
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
+  }
+
+  public void testStandardPlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new StandardPlan("standard"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
+  }
+
+  public void testManualPlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new ManualPlan("manual"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.ACTIVE);
+  }
+
+  public void testDelegatePlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new DelegatedPlan(500, 500), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
```

Comment:
I believe the reason it works for `ManualPerformance` is because of:
```
    if (performer instanceof ManualPerformance) {
      activeManualPerformances.add((ManualPerformance) performer);
      notifyTargetStateChanged();
    }
```
but `notifyTargetStateChanged` which in turn triggers the call into `setTargetState` back in the `Scheduler`.